### ISSUE TITLE
Nodeport is already allocated

### ISSUE THREAD

**Issue by roylkng** [kubernetes/kubernetes#114911]
### What happened?

On creating following service

apiVersion: v1
kind: Service
metadata:
  name: coordinator-mimic
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: hke-mimic
  ports:
    - port: 25025
      targetPort: 8000
      nodePort: 32444
`

Deleted it and created it again getting error
`The Service "coordinator-mimic" is invalid: spec.ports[0].nodePort: Invalid value: 32448: provided port is already allocated`
Verified port does not shows up as in use and lb in aws console is deleted.

![image](https://user-images.githubusercontent.com/9080276/211250746-301c9013-635a-4774-9884-14ffb46c61e3.png)

### What did you expect to happen?

It should be able to create service again and nodeport should get freed and become reusable.

### How can we reproduce it (as minimally and precisely as possible)?

mimic.yaml
`
apiVersion: v1
kind: Service
metadata:
  name: coordinator-mimic
spec:
  type: LoadBalancer
  selector:
    app.kubernetes.io/name: hke-mimic
  ports:
    - port: 25025
      targetPort: 8000
      nodePort: 32444
`
      
kubectl apply -f mimic.yaml
kubectl delete -f mimic.yaml
kubectl apply -f mimic.yaml
apply after delete errors out.

### Anything else we need to know?

I check 

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```
1.23
</details>

### Cloud provider

<details>
AWS EKS
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```
 Amazon Linux 2   5.4.226-129.415.amzn2.x86_64
</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
Docker
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by uablrek** [comment]
I suspect that this is an AWS problem, not a K8s problem. I will try to re-create this outside AWS.

**Comment by uablrek** [comment]
I can't re-create this on a local K8s cluster;

```
# kubectl get svc alpine
NAME     TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
alpine   LoadBalancer   12.0.199.250   <pending>     5001:32444/TCP   64s
# kubectl delete -f $yamld/svc.yaml; kubectl apply -f $yamld/svc.yaml
service "alpine" deleted
service/alpine created
# kubectl get svc alpine
NAME     TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)          AGE
alpine   LoadBalancer   12.0.150.20   <pending>     5001:32444/TCP   1s
```
The svc is deleted and re-created back-to-back without problems.

<details><summary>service manifest</summary>

```yaml
apiVersion: v1
kind: Service
metadata:
  name: alpine
spec:
  ipFamilyPolicy: RequireDualStack
  selector:
    app: alpine
  type: LoadBalancer
  ports:
  - port: 5001
    name: nc
    nodePort: 32444
```

</details>

In a public cloud, like AWS, things work differently. The coud-provider watches services with `type: LoadBalancer` and set's up external load balancing.

But this is not handled by Kubernetes, and thus not anything that can be fixed by any modification in K8s. So I close this issue as it is not a K8s bug.

/close

**Comment by uablrek** [comment]
@roylkng I should have checked cloud-providers more closely before answering. Sorry about that.

Checking a bit more I found that there may be a way to use `finalizers` in the service to prevent this problem. I only found this for Azure;
https://learn.microsoft.com/en-us/answers/questions/630765/aks-cluster-stop-deleting-load-balancer-event.html

But AWS probably have something similar. The finalizer should stall the delete of the service object until the cloud-provider load-balancer has been cleared. When the service is re-created it should get a new external address (which may be what you want?).
